### PR TITLE
Add support for one-time setup and also passing Thread and Invocation…

### PR DESCRIPTION
**Pull request by jlhughes**
… count to the benchmark

### CODE REVIEW

File: src/foam/nanos/bench/BenchmarkResultSystemDAO.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ foam.CLASS({
 
       br.setCores(Runtime.getRuntime().availableProcessors());
 
+      Runtime.getRuntime().gc();
```
Comment (kgrgreer) [inline src/foam/nanos/bench/BenchmarkResultSystemDAO.js:41]: It used to be that you had to do this twice, but I'm not sure that it actually does anything anymore. But better to call twice, just incase.

File: src/foam/nanos/bench/BenchmarkRunner.js [hunk 2]
```diff
@@ -119,6 +137,11 @@ foam.CLASS({
       name: 'clearPMs',
       docmentation: 'clear PMs before executing the benchmark',
     },
+    {
+      class: 'Boolean',
+      name: 'oneTimeSetup',
+      value: false
```
Comment (kgrgreer) [inline src/foam/nanos/bench/BenchmarkRunner.js]: redundant as false is the default default anyway

File: src/foam/nanos/bench/BenchmarkResultSystemDAO.js (hunk 1 above)
Comment (jlhughes) [reply src/foam/nanos/bench/BenchmarkResultSystemDAO.js:41]: @kgrgreer found this which I think describes the issues. So it's not that two calls are needed but multiple calls to trigger subsequent gc stages.   
"Directly calling System.gc() is always dubious but I can understand why people have seen that multiple calls are required to clean everything up. I don't know if it's still the case but it used to be that objects with finalizers were put in a finalization queue after being garbage collected then the finalizers were actually invoked on the next GC execution. If the objects were still eligible for garbage collection then they were finally deleted after that."

Comment (kgrgreer) [reply src/foam/nanos/bench/BenchmarkResultSystemDAO.js:41]: Also, when they used a Mark & Sweep GC, you needed two passes, one for the mark and one for the sweep to actually GC things. I'm not sure of the actual reason, but from past experiments, I could see that this was the case, but I'm not sure if this has been the case for a long time.

**Pull request merged**